riot_pal/dut_pyshell.py

```diff
@@ -0,0 +1,258 @@
+#! /usr/bin/env python3
+# Copyright (c) 2019 Kevin Weiss, for HAW Hamburg  <kevin.weiss@haw-hamburg.de>
+#
+# This file is subject to the terms and conditions of the MIT License. See the
+# file LICENSE in the top level directory for more details.
+# SPDX-License-Identifier:    MIT
+
+"""
+This script handles interfacing to the PHiLIP device.  It parses data and
+exposes from the PhilipExtIf class to be run in a shell.
+
+The purpose of this script is allow easy setup and manual usage of PHiLIP.
+
+Usage
+-----
+
+```
+usage: dut_pyshell.py   [-h]
+                        [--loglevel {debug,info,warning,error,fatal,critical}]
+                        [--port PORT]
+                        [--filter-data]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --loglevel {debug,info,warning,error,fatal,critical}
+                        Python logger log level (default: warning)
+  --port, -p
+                        Specify the serial port
+  --data_only, -do
+                        Filters data from philip responses to only display
+                        what is needed (default: True)
+```
+"""
+import cmd
+from json import dumps
+import logging
+import argparse
+import serial.tools.list_ports
+try:
+    import readline
+except ImportError:
+    readline = None
+try:
+    from .dut_shell import DutShell, RESULT_SUCCESS
+except ImportError:
+    from dut_shell import DutShell, RESULT_SUCCESS
+
+
+class DutPyShell(cmd.Cmd):
+    """Command loop for the PHiLIP interface
+
+    Args:
+        port - Serial port for the PHiLIP, if None connection wizard tries to
+               connent
+        data_only - If true only data prints from command an not the whole
+                    response struct
+    """
+    prompt = 'node: '
+
+    def __init__(self, port=None, data_only=True):
+        if port is None:
+            self.dut = self._connect_wizard()
+        else:
+            self.dut = DutShell(port, parser='json')
+        self.cmd_list = self.dut.send_cmd('help')['data']
+        self.data_only = data_only
+        cmd.Cmd.__init__(self)
+
+    @staticmethod
+    def _connect_wizard():
+        print("Starting DUT python shell")
+        serial_devices = serial.tools.list_ports.comports()
+        if len(serial_devices) == 0:
+            raise ConnectionError("Could not find any available devices")
+        elif len(serial_devices) == 1:
+            print('Connected to {}'.format(serial_devices[0]))
+            return DutShell(port=serial_devices[0][0], parser='json')
+        else:
+            print('Select a serial port:')
+            for i, s_dev in enumerate(serial_devices):
+                print("{}: {}".format(i, s_dev))
+            s_num = input("Selection(number): ")
```

Comment:
this and the next line are not very safe, I think something like this might be better:

```Python
try:
    s_num = int(input("Selection(number): "))
except:
    #error message?
    raise ValueError
```

Reply:
Agreed